File: JAVA/lecture6/interface/src/Point.java
public class Point implements Control
{
    public Point(int x, int y) {
        setX(x);
        setY(y);
    }

    @Override
    public void zkontroluj()
    {
        if ((x <= 0) || (y <= 0)) {
            System.out.println("Point is not OKAY:/");
        }
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    private int x;
    private int y;
}

File: PRC1/exercise/exercise3/point/point.cpp
#include "point.h"
#include <iostream>
#include <cstdlib>

Point::Point()
{
    std::cout << "Constructor without parameters\n";
    setX(0);
    setY(0);
}

Point::Point(double x, double y)
    : x(x), y(y)
{
    std::cout << "Constructor with two parameters\n";
}

Point::~Point()
{
    std::cout << "Destructor!\n";
}

double Point::distance(Point &other) const
{
    return sqrt(pow(this->x - other.x, 2) + pow(this->y - other.y,2));
}

void Point::print() const
{
    std::cout << "[" << this->getX() << ", " << getY() << "]\n";
}


void print(Point &p)
{
    p.print();
    // cout << "[" <<  p.getX() << ", " << p.getY() << "]" << endl;
}

float distance(Point &p1, Point &p2)
{
    return p1.distance(p2);
}